Add "Quick start" guide (#3801)
spantaleev/matrix-docker-ansible-deploy: docs/quick-start.md

## Create your user account

<sup>This section is optimized for this quick-start guide and is derived from the following full-documentation page: [Registering users](registering-users.md)</sup>

As you have configured your brand new server and the client, you need to **create your user account** on your Matrix server.

To register a user via this Ansible playbook, run the command below on your local computer.

**Notes**:
- Before running it, make sure to edit `YOUR_USERNAME_HERE` and `YOUR_PASSWORD_HERE`
- In the command below, `YOUR_USERNAME_HERE` is just a plain username (like `john`), not your full `@user:example.com` identifier

```sh
ansible-playbook -i inventory/hosts setup.yml --extra-vars='username=YOUR_USERNAME_HERE password=YOUR_PASSWORD_HERE admin=<yes|no>' --tags=register-user

# Example: ansible-playbook -i inventory/hosts setup.yml --extra-vars='username=john password=secret-password admin=yes' --tags=register-user
```

## Finalize server installation

<sup>This section is optimized for this quick-start guide and is derived from the following full-documentation page: [Server Delegation](howto-server-delegation.md)</sup>

> Now you've configured Matrix services and your user account, you need to **finalize the installation process** by [setting up Matrix delegation (redirection)](howto-server-delegation.md), so that your Matrix server (`matrix.example.com`) can present itself as the base domain (`example.com`) in the Matrix network.

Reviewer: ```suggestion
Now that you've configured Matrix services and your user account, you need to **finalize the installation process** by [setting up Matrix delegation (redirection)](howto-server-delegation.md), so that your Matrix server (`matrix.example.com`) can present itself as the base domain (`example.com`) in the Matrix network.
```

**This is required for federation to work!** Without a proper configuration, your server will effectively not be part of the Matrix network.

> There are several ways to configure the delegation, but the easiest one is to [serve the base domain](configuring-playbook-base-domain-serving.md) from the integrated web server.

Reviewer: It feels like this section focuses on the "base domain serving" option, which is nice and quick, but..

People who cannot afford to point the base domain to the playbook will need other options.

Above, we already link to `howto-server-delegation.md` twice, but maybe we should also be more explicit as to what your options are if you cannot serve the base domain on the Matrix server.

Perhaps a sentence like this can be added somewhere:

> If you're using the base domain for other purposes and cannot point it to the Matrix server (and thus cannot "serve the base domain" from it), you most likely need to [Manually install well-known files on the base domain's server](configuring-well-known.md#manually-installing-well-known-files-on-the-base-domains-server), but feel free to familiarize yourself with all [Server delegation (redirection)](howto-server-delegation.md) options.

-------

More restructuring could be done as well. For example (just a quick draft below):

> You have 2 options:
>
> - If you can afford to point the base domain at the Matrix server, follow the instructions below which guide you into [serving the base domain](configuring-playbook-base-domain-serving.md) from the Matrix server
> - Alternatively, if you're using the base domain for other purposes and cannot point it to the Matrix server (and thus cannot "serve the base domain" from it), you most likely need to [Manually install well-known files on the base domain's server](configuring-well-known.md#manually-installing-well-known-files-on-the-base-domains-server), but feel free to familiarize yourself with all [Server delegation (redirection)](howto-server-delegation.md) options.

To enable it, add the following configuration to your `inventory/host_vars/matrix.example.com/vars.yml` file:

```yaml
matrix_static_files_container_labels_base_domain_enabled: true
```

After configuring the playbook, run the installation command and wait until it finishes:

```sh
ansible-playbook -i inventory/hosts setup.yml --tags=install-all,start
```

After the command finishes, you can also check whether your server federates with the Matrix network, by using the [Federation Tester](https://federationtester.matrix.org/).

If you think something is off with the server configuration, feel free to re-run the setup command any time.

## Log in to your user account

Finally, let's make sure that you can log in to the created account with the specified password.

> You should be able to log in to it with your own Element Web client which you have set up at `element.example.com` by running the playbook. Open the URL (`https://element.example.com`) on a web browser, and enter the credentials to log in.

Reviewer: ```suggestion
You should be able to log in to it with your own [Element Web](configuring-playbook-client-element-web.md) client which you have set up at `element.example.com` by running the playbook. Open the URL (`https://element.example.com`) in a web browser and enter your credentials to log in.
```

The main suggestion here is `on` -> `in`, but I felt like adding a link for additional reading is always helpful.

> **If you successfully logged in to your account, installing and configuring procedure is complete**🎉

Reviewer: ```suggestion
**If you successfully logged in to your account, installing and configuring is complete**🎉
```

### Run installation command

Then, run the command below to start installation:

````sh
ansible-playbook -i inventory/hosts setup.yml --tags=install-all,ensure-matrix-users-created,start
````

If you **don't** use SSH keys for authentication, but rather a regular password, you may need to add `--ask-pass` to the command.

If you **do** use SSH keys for authentication, **and** use a non-root user to *become* root (sudo), you may need to add `-K` (`--ask-become-pass`) to the command.

Wait until the command completes. If it's all green, everything should be running properly.

## Create your user account

<sup>This section is optimized for this quick-start guide and is derived from the following full-documentation page: [Registering users](registering-users.md)</sup>

As you have configured your brand new server and the client, you need to **create your user account** on your Matrix server.

To register a user via this Ansible playbook, run the command below on your local computer.

**Notes**:
- Make sure to adjust `YOUR_USERNAME_HERE` and `YOUR_PASSWORD_HERE`
- For `YOUR_USERNAME_HERE`, use a plain username like `john`, not your full identifier (`@user:example.com`)

> - Use `admin=yes` or `admin=no` depending on whether you wish to make the user an administrator of the Matrix server

Author: I have two questions here:

- Is it clear which effect making a user an admin of the Matrix server has? (I guess you should never make a user admin casually)
- Should you make yourself (= the first user of the Matrix server) an administrator? Is it rather optional?

Reviewer: This section is named "Create your user account" which may not be great, because you may wish to create multiple accounts here. While it's possible, it doesn't seem obvious.

Yes, you will most likely wish to create your first user account as an administrator. The person who sets up the Matrix server will definitely wish to create their account as an administrator.

You're right that we don't explain what the effect is and that one shouldn't make users administrators willy-nilly.

Author: > This section is named "Create your user account" which may not be great, because you may wish to create multiple accounts here. While it's possible, it doesn't seem obvious.

I would add: `For an instruction to add multiple user accounts, see [Registering users](docs/registering-users.md)`, because creating other accounts than your user account is out of the scope of this document, whose objects are listed at the top.

To configure the delegation, you have these two options. Choose one of them according to your situation.

- If you can afford to point the base domain at the Matrix server, follow the instruction below which guides you into [serving the base domain](configuring-playbook-base-domain-serving.md) from the integrated web server.
- Alternatively, if you're using the base domain for other purposes and cannot point it to the Matrix server (and thus cannot "serve the base domain" from it), you most likely need to [manually install well-known files on the base domain's server](configuring-well-known.md#manually-installing-well-known-files-on-the-base-domains-server).

To have the base domain served from the integrated web server, add the following configuration to your `inventory/host_vars/matrix.example.com/vars.yml` file:

```yaml
matrix_static_files_container_labels_base_domain_enabled: true
```

After configuring the playbook, run the installation command and wait until it finishes:

```sh
ansible-playbook -i inventory/hosts setup.yml --tags=install-all,start
```

> After the command finishes, you can also check whether your server federates with the Matrix network, by using the [Federation Tester](https://federationtester.matrix.org/).

Reviewer: ```suggestion
After the command finishes, you can also check whether your server federates with the Matrix network, by using the [Federation Tester](https://federationtester.matrix.org/) and running a check using your base domain (`example.com`), not the `matrix.example.com` subdomain.
```

We often have people doing this wrong - using `matrix.example.com` in the Federation Tester. It's probably good to explicitly clarify.

It might take some time for the DNS records to propagate after creation.

> **💡 Note**: if you are using Cloudflare DNS, make sure to disable the proxy and set all records to "DNS only"

Author: Off-topic (as a memo): shall we create a emoji guide for consistency on a style guide or something?

- Warning emoji with `**Warning**:`
- Light bulb emoji with `**Note(s)**:`

We might need some time for consideration.
